statscache_plugins/volume/by_category.py

```diff
@@ -14,28 +13,31 @@ class PluginMixin(VolumePluginMixin):
     the bus for each category.
     """
 
-    def handle(self, session, messages):
-        volumes = collections.defaultdict(int)
-        for msg in messages:
-            msg_timestamp = datetime.datetime.fromtimestamp(msg['timestamp'])
-            volumes[(msg['topic'].split('.')[3],
-                     self.frequency.next(now=msg_timestamp))] += 1
+    def process(self, message):
+        timestamp = self.schedule.next(
+            now=datetime.datetime.fromtimestamp(message['timestamp'])
+        )
+        category = message['topic'].split('.')[3]
+        self._volumes[(category, timestamp)] += 1
 
-        for key, volume in volumes.items():
+    def update(self, session):
```

Comment:
It may be that these ``update`` methods can be simplified even more.

Lots of their logic is about detecting previous values stored at the 'current' timestamp by some previous partial run, and then incrementing that stored value.

Now, plugins are keeping their running queue (``self._volumes`` in this case) in memory, and then only flushing it out to disk later.

We had previously off-loaded this in-memory state to the DB to be able to recover from unexpected crashes, but it seems that the central statscache bits are now responsible for figuring out the missed workload on behalf of plugins, freeing the plugins themselves of this responsibility.